crates/libs/bindgen/src/extensions/variant_bool.rs

```diff
@@ -0,0 +1,79 @@
+use super::*;
+
+pub fn gen() -> TokenStream {
+    quote! {
+        impl VARIANT_BOOL {
+            #[inline]
+            pub fn as_bool(self) -> bool {
+                self.0 != 0
+            }
+            #[inline]
+            pub fn ok(self) -> ::windows::core::Result<()> {
+                if self.as_bool() {
+                    Ok(())
+                } else {
+                    Err(::windows::core::Error::from_win32())
+                }
+            }
+            #[inline]
+            #[track_caller]
+            pub fn unwrap(self) {
+                self.ok().unwrap();
+            }
+            #[inline]
+            #[track_caller]
+            pub fn expect(self, msg: &str) {
+                self.ok().expect(msg);
+            }
+        }
+        impl ::core::convert::From<::core::option::Option<VARIANT_BOOL>> for VARIANT_BOOL {
+            fn from(optional: ::core::option::Option<Self>) -> Self {
+                optional.unwrap_or_default()
+            }
+        }
```

Comment:
Hmm, why was this added? I'm just comparing against `BOOL` (in `bool32.rs`) which does not have this conversion.

Reply:
I added this because it is boolean functionality that was derived for `BOOL` and `BOOLEAN`, but not derived for `VARIANT_BOOL`. Even though this functionality is valid functionality for `VARIANT_BOOL`.

Reply:
This is why I asked my question earlier. https://github.com/microsoft/windows-rs/pull/2206#issuecomment-1332352847

Reply:
Oh, I see. That's because (for unknown reasons) the `BOOL` type is defined as a `struct` by the Win32 metadata whereas the `VARIANT_BOOL` type is defined as an `enum`. We can raise this discrepancy on the https://github.com/microsoft/win32metadata/ repo, but for now let's just make the extension code consistent.